src/test/java/org/fogbowcloud/manager/core/ApplicationFacadeTest.java

```diff
@@ -12,1701 +12,2209 @@
+        this.aaaController = Mockito.mock(AaController.class);
+
+        HomeDir.getInstance().setPath("src/test/resources/private");
+
+        DatabaseManager databaseManager = Mockito.mock(DatabaseManager.class);
+        Mockito.when(databaseManager.readActiveOrders(OrderState.OPEN))
+                .thenReturn(new SynchronizedDoublyLinkedList());
+        Mockito.when(databaseManager.readActiveOrders(OrderState.SPAWNING))
+                .thenReturn(new SynchronizedDoublyLinkedList());
+        Mockito.when(databaseManager.readActiveOrders(OrderState.FAILED))
+                .thenReturn(new SynchronizedDoublyLinkedList());
+        Mockito.when(databaseManager.readActiveOrders(OrderState.FULFILLED))
+                .thenReturn(new SynchronizedDoublyLinkedList());
+        Mockito.when(databaseManager.readActiveOrders(OrderState.PENDING))
+                .thenReturn(new SynchronizedDoublyLinkedList());
+        Mockito.when(databaseManager.readActiveOrders(OrderState.CLOSED))
+                .thenReturn(new SynchronizedDoublyLinkedList());
+
+        Mockito.doNothing().when(databaseManager).add(Matchers.any(Order.class));
+        Mockito.doNothing().when(databaseManager).update(Matchers.any(Order.class));
+
+        PowerMockito.mockStatic(DatabaseManager.class);
+        BDDMockito.given(DatabaseManager.getInstance()).willReturn(databaseManager);
+
+        this.orderController = Mockito.spy(new OrderController());
+        this.application = Mockito.spy(ApplicationFacade.getInstance());
+        this.application.setAaController(this.aaaController);
+        this.application.setOrderController(this.orderController);
+
+        SharedOrderHolders sharedOrderHolders = SharedOrderHolders.getInstance();
+        this.activeOrdersMap = sharedOrderHolders.getActiveOrdersMap();
+    }
 
+    // test case: When calling the method changeNetworkOrderIdsToNetworInstanceIds(), it must change
+    // the collection of OrderIDs by NetworkInstacesIDs.
     @Test
     public void changeNetworkOrderIdsToNetworInstanceIdsTest() {
+
+        // set up
         NetworkOrder networkOrder = Mockito.mock(NetworkOrder.class);
 
-        Mockito.doReturn("fake-order-id").when(networkOrder).getId();
-        Mockito.doReturn("fake-instance-id").when(networkOrder).getInstanceId();
+        Mockito.doReturn(FAKE_ORDER_ID).when(networkOrder).getId();
+        Mockito.doReturn(FAKE_INSTANCE_ID).when(networkOrder).getInstanceId();
 
         this.activeOrdersMap.put(networkOrder.getId(), networkOrder);
 
         List<String> expectedList = new ArrayList<>();
-        expectedList.add("fake-order-id");
+        expectedList.add(FAKE_ORDER_ID);
 
         ComputeOrder computeOrder = Mockito.mock(ComputeOrder.class);
         Mockito.doReturn(expectedList).when(computeOrder).getNetworksId();
 
+        // exercise
         this.application.changeNetworkOrderIdsToNetworInstanceIds(computeOrder);
 
+        // verify
```

Comment:
I think we cannot mock the return of the computerOrder.getNetworkId. This is the behaviour we want to check. One option is to not use a mock for it or to use and to check we called the set method

Reply:
Will be fixed at the next pull request...